### PR TITLE
Add vSphere IPI

### PR DESCRIPTION
**Pull request by jcpowermac**

### CODE REVIEW

File: enhancements/installer/vsphere-ipi.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+- Does vsphere need any metadata changes:
+  - `pkg/asset/cluster/vsphere`
+  - `pkg/asset/cluster/vsphere/vsphere.go`
+- Do we need to create resource pools?
```
Comment (mtnbikenc) [inline enhancements/installer/vsphere-ipi.md]: https://issues.redhat.com/browse/CORS-1329

**Comment by jcpowermac** [comment]
@sdodson 
@abhinavdahiya 
What are we critically missing from this?

@dav1x when you get a chance if you can take a look as well.

File: enhancements/installer/vsphere-ipi.md [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+on the [spike](https://issues.redhat.com/browse/CORS-1295):
+  - The installer will use the baseURI and images.vmware.path from rhcos.json to download the published OVA.
+  - For disconnected, the user can provide in the install-config.yaml a full URI path (file or URL) to the OVA which the user has previously downloaded and made available.
+  - The installer will import the OVA and create a VM template and include the infraID in the name. The VM template will be placed in the root of the vSphere hierarchy.
+  - All created entities should include the infraID to allow cleanup when running 'destroy'.
+  - Out of scope:
+    - Custom placement of the VM template
+    - Custom name of the VM template
+  - `pkg/rhcos/vsphere.go`
+- Add machines and machinesets
```
Comment (enxebre) [inline enhancements/installer/vsphere-ipi.md]: Developing for the machine API is happening here
API 
https://github.com/openshift/machine-api-operator/tree/master/pkg/apis/vsphereprovider
Controller
https://github.com/openshift/machine-api-operator/tree/master/pkg/controller/vsphere

We'll start by supporting only vCenter interactions (no raw esxi) and only supporting cloning templates on "creation". We'll extend this behaviour later on only if there's a product requirement.

We are assuming the existence of a secret with a user and password keys used to login against the vCenter https://github.com/openshift/machine-api-operator/blob/master/pkg/controller/vsphere/machine_scope.go#L57

**Comment by russellb** [comment]
@jcpowermac I know vSphere IPI support has already merged.  Can you help make sure this enhancement is updated to reflect what has been done and then we'll merge it to reflect the current state of things?  Thanks!

**Comment by sdodson** [comment]
> @jcpowermac I know vSphere IPI support has already merged. Can you help make sure this enhancement is updated to reflect what has been done and then we'll merge it to reflect the current state of things? Thanks!

@jcpowermac bump on this, can we do a quick review to make sure there aren't significant deviations between this enhancement and what's actually been implemented and then merge it?

**Comment by russellb** [comment]
@jcpowermac bump.  :-)  (also pinged on slack)

**Comment by russellb** [comment]
@jcpowermac says this is close enough and that the CI section is the most out of date.  We should just merge as is.

/lgtm

**Pull request merged**